packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,269 @@
+                    # Remember the name for later processing
+                    repo=RhsmRepository(self.module, _name=value)
+                    repos.append(repo)
+                    repos[-1].__setattr__(key, value)
+                elif repos:
+                    # Associate value with most recently recorded repo
+                    repos[-1].__setattr__(key, value)
+             
+        return repos
+
+    def filter(self, state=None):
+        ''' 
+            Return a list of Repositories where state matches (null: all, enabled, disabled)
+        '''
+
+        if self.repos is None:
+            self.repos = self._load_repo_list()
+
+            if state == 'all' or state is None:
+                for repo in self.repos:
+                    yield repo 
+
+            elif state == 'enabled':
+                for repo in self.repos:
+                    if repo.is_enabled():
+                        yield repo
+
+            elif state == 'disabled':
+                for repo in self.repos:
+                    if repo.is_disabled():
+                        yield repo
+
+    def enablerepo(self, name):
+        if len(name) > 0:
+            args="subscription-manager repos "
+            for n in name:
+                args+=" --enable="+n
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+    def disablerepo(self, name):
+        if len(name) > 0:
+            args="subscription-manager repos "
+            for n in name:
+                args += " --disable=" + n 
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+    def disable_all(self):
+        args = "subscription-manager repos --disable=*"
+        rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+def list_stuff(rhsmrepos, stuff):
+    l=[]
+    for repo in rhsmrepos.filter(stuff):
+        l.append(dict(name=repo._name, enabled=repo.is_enabled))
+    return l
+
+def main():
+    #
+    # Initialise Ansible module
+    #
+    module = AnsibleModule(
+                argument_spec = dict(
+                    name    = dict(default=None, required=False, type="list"),
+                    state   = dict(default=None, choices=['enabled','disabled']),
+                    list    = dict(default=None, required=False),
+                ),
+                required_one_of = [['name', 'list']],
+                mutually_exclusive = [['name', 'list']],
+                supports_check_mode = False,
+            )
+
+    #
+    # Initialize RhsmRepositories
+    #
+    rhsmrepos = RhsmRepositories(module)
+
+    list = module.params['list']
+    name = module.params['name']
+    state = module.params['state']
+    #
+
+    if list:
+        result=list_stuff(rhsmrepos, state)         # FIXME : maybe review state usage here
+        module.exit_json(changed=False,repos=result)
+
+    elif name is not None:
+        if name == '*':
+            rhsmrepos.disable_all()
```

Comment:
@jfenal isn't this an issue? doesn't this mean that if you specify * as the name then it will ignore the specified state and always do a disable? shouldn't it still check the state if * is specified?